src/v/rpc/connection_cache.cc

```diff
@@ -331,10 +331,12 @@ ss::future<> connection_cache::reset_client_backoff(
         return ss::now();
     }
 
-    return container().invoke_on(
-      *shard, [node_id](rpc::connection_cache& cache) mutable {
-          cache._cache.reset_client_backoff(node_id);
-      });
+    return ss::with_gate(_gate, [this, node_id, shard] {
```

Comment:
I've added checks to ensure that the gate isn't closed before accessing data members. On async functions I've also ensured that the gate is held until after all futures have completed. This should avoid any races during the destruction of `rpc::connection_cache`.

Reply:
thanks. seems we could have grabbed the gate on the destination core as an alteratnive.

Reply:
Definitely, right now the gate is only grabbed if the method is going to schedule some async task. So if a method doesn't schedule any async tasks it just checks if the gate is open and moves on. This could cause an issue if some other developer later changes the method to schedule an async task. So I see your point I think.